host/storage/storage.go

```diff
@@ -850,6 +857,18 @@ func (vm *VolumeManager) Read(root types.Hash256) (*[rhpv2.SectorSize]byte, erro
 	}
 	defer done()
 
+	vm.mu.Lock()
+	defer vm.mu.Unlock()
+
+	// Check the cache first
+	if sector, ok := vm.cache.Get(root); ok {
+		atomic.AddUint64(&vm.cacheHits, 1) // Increment cache hit counter
+		// Remove sector from cache to simulate cache miss
+		vm.cache.Remove(root)
```

Comment:
We don't want to remove the sector from the cache here, especially not for a test. A better way would be to add `VolumeManager.ResizeCache(int)` and set the cache size to 0 in the broken tests. There should only be two broken ones: `TestStoreSector` and `TestCheckIntegrity`. 

- In `TestCheckIntegrity`, you have direct access to the storage manager. 
- In `TestStoreSector`, you can get it through `host.Storage()`